api/app/utils.py

```diff
@@ -0,0 +1,16 @@
+from ast import literal_eval
+
+import requests
+from fastapi import HTTPException
+
+
+def forward_http_error(resp: requests.Response, excluded_codes: list[int]) -> None:
+    if not resp.status_code > 399:
+        return
+
+    if excluded_codes and resp.status_code in excluded_codes:
+        raise HTTPException(status_code=500, detail="An internal error occurred.")
+    else:
+        obj = resp.json()
+        detail = obj.get("detail", "Unknown error")
```

Comment:
Do we have a way to forward more information? Ideally, we would want to make the API error message a bit more useful when possible. What do you think?